### PR TITLE
Moved from Google Analytics to Firebase Analytics

### PR DESCRIPTION
**Pull request by jguerinet**
A required step to start using Firebase for the config stuff, and because Google Analytics for mobile is shutting down later this year. 
* Relevant libraries updated
* Changed GA events to FA events (removed some useless ones)
* Removed the "Send Anonymous Stats" switch in the settings (now it's just on)
* Removed some old files that were no longer being used

The events I'm sending with FA will surely be updated later to be more useful. For now I just did a simple migration, but once I play around with the console I'm sure the events will evolve.

### CODE REVIEW

File: app/src/main/java/ui/settings/about/PersonAdapter.kt [hunk 1]
```diff
@@ -107,13 +105,16 @@ class PersonAdapter : BaseRecyclerViewAdapter(), KoinComponent {
 
                 // LinkedIn
                 linkedIn.setOnClickListener {
-                    ga.sendEvent("About", "LinkedIn", person.name)
+                    FirebaseAnalytics.getInstance(it.context)
```
Comment (AllanWang) [inline app/src/main/java/ui/settings/about/PersonAdapter.kt]: If you want you can pass the context to PersonAdapter and get the instance once directly

**Comment by AllanWang** [comment]
See:

https://stackoverflow.com/questions/39813710/calling-firebase-analytics-getinstance-every-time-vs-storing-instance-as-a-s

https://stackoverflow.com/questions/43605764/will-it-leak-firebaseanalytics-getinstance-in-custom-singleton/43614370#43614370

**Comment by jguerinet** [comment]
>If you can, you may also consider loading it once using the application context, and keeping it alive during the entire lifecycle of the app

This comment makes a bit more sense now (see last point [here](https://github.com/jguerinet/MyMartlet/pull/109#issuecomment-462808158)). So again, do you think that having `Firebase` be injected everywhere by Koin as opposed to using its singleton static methods would be better? Maybe from a testing perspective in the future, this will allow us to mock it?

**Comment by AllanWang** [comment]
> > If you can, you may also consider loading it once using the application context, and keeping it alive during the entire lifecycle of the app
> 
> This comment makes a bit more sense now (see last point [here](https://github.com/jguerinet/MyMartlet/pull/109#issuecomment-462808158)). So again, do you think that having `Firebase` be injected everywhere by Koin as opposed to using its singleton static methods would be better? Maybe from a testing perspective in the future, this will allow us to mock it?

We can reply in the other thread since it's basically duplicated. I think koin with a generic interface is better so you can mock the events, but if you aren't changing the interface then you should use a singleton.

**Comment by jguerinet** [comment]
Recent Changes: 
* Updated Suitcase to v5.0.1 (includes the Analytics interface change) 
* Changed from using a `FirebaseAnalytics` instance everywhere to an `Analytics` instance, which is then given a Firebase implementation by Koin. 
* Updated MaterialDialogs to v2.0.0 (comes with Suitcase v5.0.1)

**Pull request merged**